examples/demo_service/service.cpp

```diff
@@ -15,82 +15,77 @@
-
-  std::string header="HTTP/1.1 200 OK \n "        \
-    "Date: Mon, 01 Jan 1970 00:00:01 GMT \n"      \
-    "Server: IncludeOS prototype 4.0 \n"                \
-    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT \n"   \
-    "Content-Type: text/html; charset=UTF-8 \n"     \
-    "Content-Length: "+std::to_string(html.size())+"\n"   \
-    "Accept-Ranges: bytes\n"          \
-    "Connection: close\n\n";
+         // .... generate more dynamic content
+         << "<p>This is improvised http, but proper stuff is in the works.</p>"
+         << "<footer><hr/>&copy; 2016, IncludeOS AS @ 60&deg; north</footer>"
+         << "</body></html>";
+
+  const std::string html = stream.str();
+
+  const std::string header
+  {
+    "HTTP/1.1 200 OK\n"
+    "Date: Mon, 01 Jan 1970 00:00:01 GMT\n"
+    "Server: IncludeOS prototype 4.0\n"
+    "Last-Modified: Wed, 08 Jan 2003 23:11:55 GMT\n"
+    "Content-Type: text/html; charset=UTF-8\n"
+    "Content-Length: "+std::to_string(html.size())+'\n'
+    "Accept-Ranges: bytes\n"
+    "Connection: close\n\n"
+  };
+
   return header + html;
 }
 
-const std::string NOT_FOUND = "HTTP/1.1 404 Not Found \nConnection: close\n\n";
+const std::string NOT_FOUND = "HTTP/1.1 404 Not Found\nConnection: close\n\n";
 
 void Service::start() {
-  // Assign a driver (VirtioNet) to a network interface (eth0)
-  // @note: We could determine the appropirate driver dynamically, but then we'd
-  // have to include all the drivers into the image, which  we want to avoid.
-  hw::Nic<VirtioNet>& eth0 = hw::Dev::eth<0,VirtioNet>();
-
-  // Bring up a network stack, attached to the nic
-  // @note : No parameters after 'nic' means we'll use DHCP for IP config.
-  inet = std::make_unique<net::Inet4<VirtioNet> >(eth0);
+  srand(OS::cycles_since_boot());
 
+  // Stack with network interface (eth0) driven by VirtioNet
+  // DNS address defaults to 8.8.8.8
   // Static IP configuration, until we (possibly) get DHCP
-  // @note : Mostly to get a robust demo service that it works with and without DHCP
-  inet->network_config( { 10,0,0,42 },      // IP
-                        { 255,255,255,0 },  // Netmask
-                        { 10,0,0,1 },       // Gateway
-                        { 8,8,8,8 } );      // DNS
-
-  srand(OS::cycles_since_boot());
+  // @note : Mostly to get a robust demo service that works with and without DHCP
+  static auto inet = net::new_ipv4_stack<>({ 10,0,0,42 },      // IP
+                                           { 255,255,255,0 },  // Netmask
+                                           { 10,0,0,1 });      // Gateway
 
   // Set up a TCP server on port 80
   auto& server = inet->tcp().bind(80);
 
-  hw::PIT::instance().on_repeated_timeout(30s, []{
-      printf("<Service> TCP STATUS:\n%s \n", inet->tcp().status().c_str());
-    });
+  // Print some useful netstats every 30 secs
+  hw::PIT::instance().on_repeated_timeout(30s, [] {
+      printf("<Service> TCP STATUS:\n\t%s\n", inet->tcp().status().c_str());
```

Comment:
The added `\t` make the text look a little off, since the following lines (`std::string TCP::status()`) lines don't got indention:
```
<Service> TCP STATUS:
	LISTENING SOCKETS:
10.0.0.42:80 0.0.0.0:0 (LISTENING)

CONNECTIONS:
Proto	Recv	Send	In	Out	Local			Remote			State
tcp4	 	 	 	 	10.0.0.42:80		10.0.0.1:35450		ESTABLISHED

```